client/src/pages/LessonHome/LessonHome.js

```diff
@@ -90,18 +95,59 @@ const LessonHome = ({ navigation }) => {
             }
 
             // Need to fetch image uri
-            // eslint-disable-next-line no-await-in-loop
-            const uri = await trackPromise(
-              downloadImageFile(
-                currentCourseId,
-                currentUnitId,
-                currentLessonId,
-                item._id,
-                fileType,
-              ),
-            )
-
-            formattedItem.imageURI = uri
+            /* eslint-disable no-loop-func */
+            // [TODO]: Add backend trackPromise()
+            downloadImageFile(
+              currentCourseId,
+              currentUnitId,
+              currentLessonId,
+              item._id,
+              fileType,
+            ).then((value) => {
+              const cloneVocabData = _.cloneDeep(formattedVocabData)
+              cloneVocabData.find(
+                (element) => element._id === formattedItem._id,
+              ).imageURI = value
+              formattedVocabData = cloneVocabData
+              setData(formattedVocabData)
+              return value
+            })
+            /* eslint-enable no-loop-func */
+          }
+
+          if (item.audioURI) {
+            formattedItem.audioURI = item.audioURI
+          } else if (item.audio !== '') {
+            const filePath = item.audio
+            const splitPath = filePath.split('.')
+
+            // Get the file type from the vocabItem's audio field
+            let fileType = 'm4a'
+
+            if (splitPath.length === 2) {
```

Comment:
Can you disable prefer-destructuring in the `.eslintrc.js` file at the root of the client folder?